transports/quic/tests/smoke.rs

```diff
@@ -68,614 +31,408 @@ async fn async_std_smoke() {
     smoke::<quic::async_std::Provider>().await
 }
 
-async fn smoke<P: Provider>() {
+#[cfg(feature = "async-std")]
+#[async_std::test]
+async fn dial_failure() {
     let _ = env_logger::try_init();
-    let mut rng = rand::thread_rng();
-
-    let mut a = create_swarm::<P>().await;
-    let mut b = create_swarm::<P>().await;
+    let mut a = create_transport::<quic::async_std::Provider>().1;
+    let mut b = create_transport::<quic::async_std::Provider>().1;
 
     let addr = start_listening(&mut a, "/ip4/127.0.0.1/udp/0/quic").await;
+    drop(a); // stop a so b can never reach it
 
-    let mut data = vec![0; 4096 * 10];
-    rng.fill_bytes(&mut data);
+    match dial(&mut b, addr).await {
+        Ok(_) => panic!("Expected dial to fail"),
+        Err(error) => {
+            assert_eq!("Handshake with the remote timed out.", error.to_string())
+        }
+    };
+}
 
-    b.behaviour_mut().add_address(a.local_peer_id(), addr);
-    b.behaviour_mut()
-        .send_request(a.local_peer_id(), Ping(data.clone()));
+#[cfg(feature = "tokio")]
```

Comment:
Seems like you are stuck to async-std tests? Why then there is a tokio test?

Reply:
What do you mean stuck to? We do test both providers in here.

Reply:
dial_failure is tested with async-std only. so does ipv4_dial_ipv6, wrong_peerid, tcp_and_quic, concurrent_connections_and_streams_async_std. endpoint_reuse is tested with tokio only.

Reply:
I didn't really give it much thought to be honest so if any test is different from what it was before, it is an accident.

I can review again to retain which test uses which runtime if you want. I don't think it is particularly important as long as we have a smoke test for both!

Reply:
I just went over it again and every test is still run with the same executor as before. `endpoint_reuse` also uses `tokio` on the current `libp2p-quic` branch so this PR doesn't change any of that!

I'd prefer if we can sort this in a follow-up :)